internal/events/handler.go

```diff
@@ -63,9 +65,11 @@ func NewEventHandlerImpl(cfg EventHandlerConfig) *EventHandlerImpl {
 }
 
 func (h *EventHandlerImpl) HandleEventBatch(ctx context.Context, batch EventBatch) {
+	objects := make([]client.Object, 0, len(batch))
```

Comment:
why not all objects handled by NKG?

for example, let's say an endpointslice gets upserted (endpoints go from X to Y for a service) and that results into NGINX reload failure.  I think in this case the Gateway also becomes not programmed.

Reply:
Any time an error is returned when reloading nginx, we'll write the status. Doesn't matter which resource triggered it.

Reply:
> Any time an error is returned when reloading nginx, we'll write the status. Doesn't matter which resource triggered it.

yep, but we're limiting the scope of the objects to the ones present in the batch, not all objects captured in the Statuses

Reply:
Oh I see what you're saying. Do we have anywhere that we store all gateway API objects that exist in the system?